cmd/prune-junit-xml/prunexml_test.go

```diff
@@ -65,3 +65,55 @@ func TestPruneXML(t *testing.T) {
+		</testcase>
+	</testsuite>
+</testsuites>`
+
+	outputXML := `<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+	<testsuite tests="6" failures="0" time="5.50000" name="k8s.io/kubernetes/cluster/gce/cos" timestamp="">
+		<properties>
+			<property name="go.version" value="go1.18 linux/amd64"></property>
+		</properties>
+		<testcase classname="k8s.io/kubernetes/cluster/gce/cos" name="TestServerOverride" time="1.610000"></testcase>
+		<testcase classname="k8s.io/kubernetes/cluster/gce/cos" name="TestStorageOptions" time="1.140000"></testcase>
```

Comment:
I agree with @BenTheElder when he wrote:
> I think we should be considering package name as the top level and nesting under that, there's already a ridiculous amount of packages and those are more stable top level identifiers

In this example, the output then would have only `<testcase classname="k8s.io/kubernetes/cluster/gce"  name="cos"` and not mention `TestServerOverride` and `TestStorageOptions`.

The time of that artificial `testcase` should be the sum of the `testcases` that it replaces.

Reply:
@pohly a part from package name being the top level identifiers, Ben also mentioned having other tests nested under that. Does that mean different from pruning?
We already have the option of "Group by Hierarchy Pattern", when the regular expression "\w+" is given it gets nested as below.
![image](https://user-images.githubusercontent.com/63038004/230072760-d628cdeb-7370-42cd-a11f-0fcd9694fb29.png)

But as the data remains the same, it doesn't reduce the number of rows and hence the difficulty in loading complete data still exists.

Hope as you mentioned above, pruning all Testcase rows and creating artificial tests with
`name` entity of testcase as package name
`time` = sum of testcase durations and
`failure` message=concatinating the messages from subtests would suffice.

Reply:
> We already have the option of "Group by Hierarchy Pattern", when the regular expression "\w+" is given it gets nested as below.

TIL. @michelle192837 can we make that a default for certain pages?

Reply:
IMO, even if this filter is set by default, it would not reduce the amount of data/number of rows. Thus the problem of the page not loading all columns wouldn't be solved. @michelle192837 Please correct me if I am wrong.

@BenTheElder Please let me know if you are fine with pruning the data and having package names displayed as top-level identifiers and have consistent rows for all the runs.

Also, I would like to mention, as this is going to be an additional flag to `prune-junit-xml` tool, only the users who wish to use can include it in their CI.

Reply:
@BenTheElder Please confirm if you are fine with pruning the data and having package names displayed as top-level identifiers and have consistent rows for all the runs.

So this would create artificial tests in the junit_xml file with
`name` entity of testcase as package name
`time` = sum of testcase durations and
`failure message`=concatinating the messages from subtests

Reply:
Sorry I was out for a week.

> IMO, even if this filter is set by default, it would not reduce the amount of data/number of rows. Thus the problem of the page not loading all columns wouldn't be solved.

Yes, but it very nice for the humans!

I'm fine with either approach, cc @liggitt @pohly